libraries/Microsoft.Bot.StreamingExtensions/ContentStream.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using Microsoft.Bot.StreamingExtensions.Payloads;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    internal class ContentStream : IContentStream
+    {
+        private readonly ContentStreamAssembler _assembler;
+        private PayloadStream _stream;
+
+        internal ContentStream(Guid id, ContentStreamAssembler assembler)
+        {
+            Id = id;
+            _assembler = assembler ?? throw new ArgumentNullException();
+        }
+
+        public Guid Id { get; private set; }
+
+        public string Type { get; set; }
+
+        public int? Length { get; set; }
+
```

Comment:
Why is this a nullable type? That's... a rough programming model.

Reply:
This was part of the design to differentiate between finite and infinite streams, I'll look into if it actually needs to be null for the later.

Reply:
In short we need the ability to tell if the value of Length is giving us honest information about the length of the stream or not. We're currently using the nullable<int> so we can call HasValue on it, and to avoid having it initialize to 0 and causing the assemblers and dis-assemblers from prematurely ending a stream. I'm weighing the option of using a negative value as a flag.